torch/_refs/__init__.py

```diff
@@ -3614,6 +3614,23 @@ def repeat(a: Tensor, *repeat_shape) -> Tensor:
     return permuted_result.reshape(target_shape)
 
 
+def _nd_to_1d(a: Tensor) -> Tensor:
+    torch._check(a.numel() > 0)
+    torch._check(a.ndim > 0)
+    torch._check(a.is_contiguous())
+    return torch.as_strided(a, [a.numel()], [1])
+
+
+def _1d_to_2d(a: Tensor, dim0, dim1) -> Tensor:
+    torch._check(a.numel() > 0)
+    torch._check(a.ndim == 1)
+    torch._check(a.is_contiguous())
```

Comment:
It is good to distinguish them (or ideally, NOT test the preconditions at all), because some preconditions are user error, and other are "oops the caller code shouldn't have called this helper", and they're all jumbled together right now

Reply:
I have eliminated the separate functions which seem unnecessary now. Hopefully, the revised code resolves some of these concerns.